File: src/main/java/io/github/rcarlosdasilva/cogen/core/handler/OpenExplorerHandler.java
package io.github.rcarlosdasilva.cogen.core.handler;

import io.github.rcarlosdasilva.cogen.core.Handler;
import io.github.rcarlosdasilva.cogen.core.Storage;
import io.github.rcarlosdasilva.cogen.utils.LogIf;
import lombok.extern.slf4j.Slf4j;

import java.io.IOException;

/**
 * @author <a href="mailto:rcarlosdasilva@qq.com">Dean Zhao</a>
 */
@Slf4j
public class OpenExplorerHandler implements Handler {

  @Override
  public void handle() {
    if (!Storage.getConfiguration().isOpenExplorer()) {
      return;
    }
    log.info("[explorer] - 开箱ing");

    String outDir = Storage.getConfiguration().getOut();
    try {
      String osName = System.getProperty("os.name");
      if (osName != null) {
        if (osName.contains("Mac")) {
          Runtime.getRuntime().exec("open " + outDir);
        } else if (osName.contains("Windows")) {
          Runtime.getRuntime().exec("cmd /c start " + outDir);
        } else {
          log.info("文件输出目录:" + outDir);
        }
      }
    } catch (IOException ex) {
      LogIf.error(log, "打开失败", ex);
    }
  }

}

File: src/main/kotlin/io/github/rcarlosdasilva/cogen/core/handler/OpenExplorerHandler.kt
package io.github.rcarlosdasilva.cogen.core.handler

import io.github.rcarlosdasilva.cogen.core.Handler
import io.github.rcarlosdasilva.cogen.core.Storage
import io.github.rcarlosdasilva.cogen.util.LogIf
import mu.KotlinLogging
import java.io.IOException

object OpenExplorerHandler : Handler {

  private val logger = KotlinLogging.logger {}

  override fun handle() {
    with(Storage.configuration!!) {
      if (isOpenExplorer) {
        return
      }

      logger.info("[explorer] - 验货ing")
      try {
        val osName = System.getProperty("os.name")
        when {
          osName?.contains("Mac") == true -> Runtime.getRuntime().exec("open " + out)
          osName?.contains("Windows") == true -> Runtime.getRuntime().exec("cmd /c start " + out)
          else -> logger.info("文件输出目录:" + out)
        }
      } catch (ex: IOException) {
        LogIf.error(logger, "打开失败", ex)
      }
    }
  }

}